addons/hr_expense/tests/test_expenses.py

```diff
@@ -118,8 +118,8 @@ def test_expense_main_flow(self):
         # Generate a payment for 'company_account' (and its move(s)) and a vendor bill for 'own_account'
         expense_sheets.action_sheet_move_create()
         self.assertRecordValues(expense_sheets, [
-            {'state': 'post', 'payment_state': 'not_paid', 'accounting_date': date(2021, 10, 10)},
-            {'state': 'done', 'payment_state': 'paid',     'accounting_date': date(2021, 10, 12)},  # Set to paid as move is posted directly
+            {'state': 'post', 'payment_state': 'not_paid',  'accounting_date': date(2021, 10, 10)},
+            {'state': 'done', 'payment_state': 'paid',      'accounting_date': date.today()},  # Set to paid as move is posted directly
```

Comment:
TLDR: avoid `today()` or add the decorator `@freeze_time("{a specific date}")` to the test method/class.

long blabla:
We should try to avoid to set `today()` in test to prevent nondeterministic error to make the test fail.
Dumb example of something that could go wrong test starts at 23h59 and finishes at 00h00 --> The date isn't the same anymore --> The test fails.

If we want to keep it, we should "freeze the date" on which the test happen with to make the test always run the same way.
To do that, we add a @freeze_time("specific-date") as a decorator to the test or even to the whole test class.
This will ensure that `today()` always returns a datetime/date/whatever at the the specific-date.

We can for example have issue with leap year. (I've heard several tests fails on the 29th :grimacing: )